src/core/include/openvino/core/descriptor/tensor.hpp

```diff
@@ -118,13 +103,16 @@ class OPENVINO_API Tensor {
     std::unordered_set<std::string> m_names;
     std::unordered_set<std::string>::const_iterator m_name_it;
     RTMap m_rt_info;
-    mutable std::atomic_bool m_shape_changed;
 
     friend OPENVINO_API std::string get_ov_tensor_legacy_name(const Tensor& tensor);
     friend OPENVINO_API void set_ov_tensor_legacy_name(Tensor& tensor, const std::string& tensor_name);
     friend void set_element_type(Tensor& tensor, const element::Type& elemenet_type);
     friend void set_tensor_type(Tensor& tensor, const element::Type& element_type, const PartialShape& pshape);
     friend class pass::ReverseShapeAndTypeInfer;
+
+private:
+    mutable bool m_shape_changed = true;
```

Comment:
```suggestion
    mutable bool m_shape_changed;
```
And initialize it in ctors. Others members also have no initialization in declaration.

Reply:
Other members get initialized with values from ctor parameter. This member should always be initialized with `true` and that's the best place for it.